### PR TITLE
[ABA 🎵 Impact] - Fallback message and Loader

### LINKED ISSUE #1908: [ABA 🎵 ] - Fallback message in case of 🚫 Fiat impact && 🚫 ABA impact
**Description**
Fallback message in case of 🚫 Fiat impact && 🚫 ABA impact 
What should we do if we can't show a fiat price impact NOR an ABA price impact?
Show some sort of warning, probably.

**Proposal**
Show a warning similar to the HighFeeWarning (underneath) on swap page and swap confirmation modal

**Alternatives**
Show a warning in the CurrencyInputPanel (where we show inputs and token dropdown)

## Screenshots
Swap page
image

Modal
image

Mouseover content
image

## Content
Mouseover content:
```
<small>
  We are unfortunately unable to calculate any price impact for this trade.
  <br />
  <br />
  <u>
    <strong>
      We strongly advise you to compare this trade across other dexes and for you to do your due diligence before
      advancing.
    </strong>
  </u>
  <br />
  <br />
  You may still move forward with this trade but there is a potentially high risk of executing an unfavourable
  execution.
</small>
```

### PR DESCRIPTION
**Pull request by W3stside**
# Summary

Closes #1908

Description && screenshots in #1908 

TODO: there are some tweaks to make:
1. ~better hiding/showing of message based on price impact - idea will be to (in another PR) make a global flag for ABA impact loading/not existing and wait for that to return before showing/hiding message~ done using fallback price impact error checking
2. ~a loader potentiatlly?~ done loader in commit

### LOADER screenshot
Screenshot 2021-11-29 at 7 50 32 PM

  # To Test
1. any chain
2. any tokens
3. enter amounts
4a. see loader in price impact spot
4. wait for message to appear (if selected token pair has no impact)
5. switch to other token with known impact
6. message warning should disappear

### CODE REVIEW

**Comment by MareenG** [comment]
Hey @W3stside, 

In generel it works and the messages shows up and disappear. For 1Inch and stake it didn't work I got the error loading price messages but for other tokens it worked as expected. 

However, I noticed that the error messages always shows up if you insert new values see video:

https://user-images.githubusercontent.com/34510341/143449683-6dd3b0f4-8b6d-4492-8fc0-b37db2069c89.mov

Also it doesn't quite work if you already have inserted a high value in the token amount field.
Example 
Choose WXDAI and AGVE insert a high switch amount and then change to 1Inch the message is not displayed

https://user-images.githubusercontent.com/34510341/143452918-00f80a8f-2cb3-4883-92f0-9e80def4e69b.mov

**Comment by elena-zh** [comment]
Hey @W3stside , I'd like to add some cases: 
1. Swap button is disabled to me when I select WXDAI-STAKE token pair
disabled
2. 'unknown' looks like a link to me, and intuitionally I want to click on it.. Is there any possibility to write 'unknown' in bold or so? 
seems like a link

**Comment by elena-zh** [comment]
Should we show this warning when we select XDAI/ETH as a token to trade? 
image

Currently, we have a task not to show fee warning when these tokens are selected #1853

**Comment by elena-zh** [comment]
It would be nice to show this warning a bit nicer in a mobile devices. See the image (iPhone 12 mini): dash is shifted on the 2nd row, looks weird
image

**Comment by alfetopito** [comment]
Tested only xDai

Looking great, loader behaving fine and giving the warning on unknown estimations like the infamous 1inch
Screenshot from 2021-11-29 11-09-11

But I did get a weird USD estimation (although amounts look ok) for DAI
Screenshot from 2021-11-29 11-14-30

File: src/custom/components/SwapWarnings/index.tsx [hunk 1]
```diff
@@ -104,14 +105,31 @@ const HighFeeWarningMessage = ({ feePercentage }: { feePercentage?: Fraction })
...
+    <small>
+      We are unfortunately unable to calculate any price impact for this trade.
+      <br />
+      <br />
+      <u>
+        <strong>We strongly advise you to do your due diligence before advancing.</strong>
+      </u>
+      <br />
+      <br />
+      You may still move forward with this order but there is a potentially high risk of executing an unfavorable trade.
```
Comment (anxolin) [inline src/custom/components/SwapWarnings/index.tsx:119]: For me this message is too alarmist. I think i suggested in some other PR something softer, but don't remember in which one

**Review by anxolin — COMMENTED** [review]
I love the loading, i think is a great solution!

I would delay the appearance fo the loadera bit maybe? at least in xDAI was loading fast and making an effect like everything was moving a lot. I was feeling it takes too much attention (maybe good, but i don't like this part a lot).


The numbers are completely weird, or this is out of this PR? 
image

image

File: src/custom/components/SwapWarnings/index.tsx (hunk 1 above)
Comment (W3stside) [reply src/custom/components/SwapWarnings/index.tsx:119]: "You may still move forward with this order but make sure the receive amounts are what you expect."

Comment (W3stside) [reply src/custom/components/SwapWarnings/index.tsx]: done 9cc6852

**Comment by W3stside** [comment]
@elena-zh @anxolin 

Screenshot 2021-11-29 at 11 50 51 PM

**Comment by W3stside** [comment]
> I love the loading, i think is a great solution!
> 
> I would delay the appearance fo the loadera bit maybe? at least in xDAI was loading fast and making an effect like everything was moving a lot. I was feeling it takes too much attention (maybe good, but i don't like this part a lot).
> 
ok something to consider, i'll look into it today

> The numbers are completely weird, or this is out of this PR? image

cause of the simple function im using, so in another pr will be fixed

**Comment by elena-zh** [comment]
Loading looks great to me. However, sometimes it runs too long. In addition, once I was able to get more than 2000% after a long loading, but then in a sec the value was changed to 14% or so. 

But I still see Swap button disabled when there is no price impact warning
image
high price impact

Or these changes are out of scope for this PR?

Again, I still see this issue https://github.com/gnosis/cowswap/pull/1916#issuecomment-979259335 . Please let me know if I need to open a separate issue for this. 

Another nitpick, is that there is a different padding between warnings on the Swap form and confirmation modals. It would be good to have the same one
swap
confirm

**Comment by elena-zh** [comment]
Hey @W3stside , now the 'Swap/ button is enabled, but a bit weird to see Price impact message in the confirmation modal. See the video: 
https://watch.screencastify.com/v/E7RbQ4fQqbnYFfK4Vixr

**Comment by W3stside** [comment]
@anxolin @gnosis/gp-frontend @gnosis/gp-qa please post merge review if necessary, tho this one is outdated from the others merged in (safe to ignore)

**Pull request merged**